Job listing

Ashish-CodeJourney opened the pull request:
- `/jobs` shows all job listing
- Jobs page has 2 sections
  - featured jobs
  - All job opportunities
  
- Schema validation is done via Zod
- When you click on `more details` on any job card, it will open an individual job page displaying all information about a particular job

Closes #91 and tasks #91 #93 #94

Ashish-CodeJourney committed:
build(dep): Install zod for schema validation

Ashish-CodeJourney committed:
feat: Job listing, schema validation and individual job details page

Ashish-CodeJourney committed:
fix: Remove browse by categoties section

suchit-sheth reviewed (changes requested):
Awesome work brother, 

Thanks a lot for contribution,

if possible, could you please create separate component for individual section so in future we dose not have to read whole long file

Ashish-CodeJourney committed:
refactor(jobs): Split job detail page into modular components

- Extract JobDetailHeader component for job header section
- Extract JobDescription component for job details and benefits
- Extract JobApplicationCard component for application actions
- Extract JobDetailsCard component for job metadata
- Extract JobCompanyCard component for company information
- Reduce main [id].astro from 450+ lines to ~90 lines

Ashish-CodeJourney committed:
refactor: Stats, and JobCard fixed

Ashish-CodeJourney committed:
feat: Featured jobs will not be duplicated in all jobs

Ashish-CodeJourney committed:
fix: All job related issues

Ashish-CodeJourney committed:
chore: Format using npm run format

3x10RaiseTo8 reviewed (changes requested):
Nav bar links inconsistent. In /jobs, the link in navbar shows /jobs/jobs

Ashish-CodeJourney commented:
> Nav bar links inconsistent. In /jobs, the link in navbar shows /jobs/jobs

<img width="531" height="57" alt="image" src="https://github.com/user-attachments/assets/6e4d234a-eb12-499a-b50a-baf9896e51a5" />

Ashish-CodeJourney committed:
fix: Change layout of job cards
